Sources/Mixpanel.m

```diff
@@ -207,28 +207,48 @@ - (void)didDebugInit:(NSString *)distinctId
 {
     NSString *debugInitCountKey = @"MPDebugInitCountKey";
     NSInteger debugInitCount = [[NSUserDefaults standardUserDefaults] integerForKey:debugInitCountKey] + 1;
-    if (debugInitCount == 1) {
-        [self sendHttpEvent:@"First SDK Debug Launch" apiToken:@"metrics-1" distinctId:distinctId];
-    }
+    [self sendHttpEvent:@"SDK Debug Launch" apiToken:@"metrics-1" distinctId:distinctId properties:@{@"Debug Launch Count": @(debugInitCount)}];
     [self checkForSurvey:distinctId debugInitCount: debugInitCount];
+    [self checkIfImplemented:distinctId debugInitCount: debugInitCount];
     [[NSUserDefaults standardUserDefaults] setInteger:debugInitCount forKey:debugInitCountKey];
     [[NSUserDefaults standardUserDefaults] synchronize];
 }
 
 - (void)checkForSurvey:(NSString *)distinctId
         debugInitCount:(NSInteger)debugInitCount
 {
-    NSString *surveyShownCountKey = @"MPSurveyShownCountKey";
-    NSInteger surveyShownCount = [[NSUserDefaults standardUserDefaults] integerForKey:surveyShownCountKey];
-    if ((debugInitCount > 10 && surveyShownCount < 1) || (debugInitCount > 20 && surveyShownCount < 2) || (debugInitCount > 30 && surveyShownCount < 3)) {
+    NSString *surveyShownDateKey = @"MPSurveyShownDateKey";
+    NSDate *surveyShownDate = [[NSUserDefaults standardUserDefaults] objectForKey:surveyShownDateKey];
+    if (!surveyShownDate || [surveyShownDate timeIntervalSinceNow] < -3600) { // only show once per day
         NSString *waveHand = @"\U0001f44b";
         NSString *thumbsUp = @"\U0001f44d";
         NSString *thumbsDown = @"\U0001f44e";
         NSString *logString = [NSString stringWithFormat:@"%@%@ Zihe & Jared here, tell us about the Mixpanel developer experience! https://www.mixpanel.com/devnps %@%@", waveHand, waveHand, thumbsUp, thumbsDown];
         NSLog(@"%@", logString);
-        surveyShownCount++;
-        [[NSUserDefaults standardUserDefaults] setInteger:surveyShownCount forKey:surveyShownCountKey];
-        [self sendHttpEvent:@"Dev NPS Survey Logged" apiToken:@"metrics-1" distinctId:distinctId properties:@{@"Survey Shown Count": @(surveyShownCount), @"Debug Launch Count": @(debugInitCount)}];
+        [[NSUserDefaults standardUserDefaults] setObject:[NSDate date] forKey:surveyShownDateKey];
+        [self sendHttpEvent:@"Dev NPS Survey Logged" apiToken:@"metrics-1" distinctId:distinctId properties:@{@"Debug Launch Count": @(debugInitCount)}];
+    }
+}
+
+- (void)checkIfImplemented:(NSString *)distinctId
+            debugInitCount:(NSInteger)debugInitCount
+{
+    NSString *hasImplementedKey = @"MPDebugImplementedKey";
+    BOOL hasImplemented = [[NSUserDefaults standardUserDefaults] boolForKey:hasImplementedKey];
+    if (!hasImplemented) {
+        NSInteger completed = 0;
+        BOOL hasTracked = [[NSUserDefaults standardUserDefaults] boolForKey:@"MPDebugTrackedKey"];
```

Comment:
nit: better define constants for those keys to avoid typos cause they've been referenced a couple of times.

Reply:
done